src/Boids.hs

```diff
@@ -173,12 +176,25 @@ zoomControl (EventKey (MouseButton WheelDown) Down _ _) world =
     world { worldViewPort = zoom (-zoomSpeed) $ worldViewPort world }
 zoomControl _ world = world
 
--- TODO: implement dragging around
+dragControl :: Event -> World -> World
```

Comment:
That is not quite true. `dragControl` requres `worldPrevPosition` which part of the world at the moment. Maybe we should combine `ViewPort` and any additional data into a new entity which could be called something like `Navigation`.